geode-core/src/main/java/org/apache/geode/management/internal/ManagementAgent.java

```diff
@@ -449,20 +463,15 @@ public synchronized void start() throws IOException {
       if (accessFile != null && accessFile.length() > 0) {
         // Rewire the mbs hierarchy to set accessController
         ReadOpFileAccessController controller = new ReadOpFileAccessController(accessFile);
-        controller.setMBeanServer(mbs);
+        if (jmxConnectorServer.getMBeanServer() == null) {
```

Comment:
Looks like besides re-arranging code, this is the only effective change here. Do we know whey in this case this mbeanServer is not null?

Reply:
If you look at the implementation of `JMXConnectorServer. setMBeanServerForwarder()`, if the MBeanServer of the connector server is non-null, it will set the MBeanServer for the forwarder, which is the `controller` in our case. We want to avoid setting the MBeanServer twice for the controller, otherwise it will throw `IllegalArgumentException`. 

Here is the code  of `JMXConnectorServer. setMBeanServerForwarder()`:
```
    public synchronized void setMBeanServerForwarder(MBeanServerForwarder mbsf)
    {
        if (mbsf == null)
            throw new IllegalArgumentException("Invalid null argument: mbsf");

        if (mbeanServer !=  null) mbsf.setMBeanServer(mbeanServer);
        mbeanServer = mbsf;
    }
```

Reply:
We know at this point `JMXConnectorServer`'s mbeanServer is definitely not null, so inside the call `setMBeanServerForwarder`, it will call `mbsf.setMBeanServer(mbeanServer)` anyway, should we just delete line 452 then?

Reply:
I looked at the history of the file, the call `jmxConnectorServer.setMBeanServerForwarder(controller);` was added recently. When that call is added, we don't need the previous line `controller.setMBeanServer(mbs);` anymore.

Reply:
Agree.